### ISSUE TITLE
Python Platform Dirs dependency issue

### ISSUE THREAD

**Issue by jonathanjsimon** [borgbase/vorta#1844]
### Description

The python platformdirs library has seen its 4.0.0 version become public release. Vorta depends on < 4.0.0 and will not run at all with 4.0.0 installed.  This can create a dependency nightmare. Manually downgrading to an older platformdirs library fixes the issue.

### Reproduction

- [X] I tried to reproduce the issue.
- [X] I was able to reproduce the issue.

### OS

EndeavourOS KDE Latest

### Version of Vorta

0.8.12

### What did you install Vorta with?

AUR

### Version of Borg

1.2.6

### Logs

```pytb
❯ vorta
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 631, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 968, in require
    needed = self.resolve(parse_requirements(requirements))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 829, in resolve
    dist = self._resolve_dist(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 875, in _resolve_dist
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (platformdirs 4.0.0 (/usr/lib/python3.11/site-packages), Requirement.parse('platformdirs<4.0.0,>=2.6.0'), {'vorta'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/vorta", line 33, in <module>
    sys.exit(load_entry_point('vorta==0.8.12', 'gui_scripts', 'vorta')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/bin/vorta", line 25, in importlib_load_entry_point
    return next(matches).load()
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 202, in load
    module = import_module(match.group('module'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/lib/python3.11/site-packages/vorta/__main__.py", line 9, in <module>
    from vorta.store.connection import init_db
  File "/usr/lib/python3.11/site-packages/vorta/store/connection.py", line 6, in <module>
    from .migrations import run_migrations
  File "/usr/lib/python3.11/site-packages/vorta/store/migrations.py", line 4, in <module>
    from .models import (
  File "/usr/lib/python3.11/site-packages/vorta/store/models.py", line 11, in <module>
    from vorta.utils import slugify
  File "/usr/lib/python3.11/site-packages/vorta/utils.py", line 18, in <module>
    from vorta.borg._compatibility import BorgCompatibility
  File "/usr/lib/python3.11/site-packages/vorta/borg/_compatibility.py", line 1, in <module>
    from pkg_resources import parse_version
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 3327, in <module>
    @_call_aside
     ^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 3302, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 3340, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 633, in _build_master
    return cls._build_from_requirements(__requires__)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 646, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 829, in resolve
    dist = self._resolve_dist(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 870, in _resolve_dist
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'platformdirs<4.0.0,>=2.6.0' distribution was not found and is required by vorta
```

**Comment by Orko79** [comment]
Hi!
Exact same issue here! After downgrading python-platformdirs from 4.0.0-1 to 3.11.0-1 vorta works fine. Added python-platformdirs to IgnorePkg until fix.

### Reproduction

[x] I tried to reproduce the issue.
[x] I was able to reproduce the issue.

### OS
EndeavourOS KDE Latest

### Version of Vorta
0.8.12

### What did you install Vorta with?
pacman from Chaotic-AUR

### Version of Borg
1.2.6

### Logs
```python
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 631, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 968, in require
    needed = self.resolve(parse_requirements(requirements))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 829, in resolve
    dist = self._resolve_dist(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 875, in _resolve_dist
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (platformdirs 4.0.0 (/usr/lib/python3.11/site-packages), Requirement.parse('platformdirs<4.0.0,>=2.6.0'), {'vorta'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/vorta", line 33, in <module>
    sys.exit(load_entry_point('vorta==0.8.12', 'gui_scripts', 'vorta')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/bin/vorta", line 25, in importlib_load_entry_point
    return next(matches).load()
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 202, in load
    module = import_module(match.group('module'))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/lib/python3.11/site-packages/vorta/__main__.py", line 9, in <module>
    from vorta.store.connection import init_db
  File "/usr/lib/python3.11/site-packages/vorta/store/connection.py", line 6, in <module>
    from .migrations import run_migrations
  File "/usr/lib/python3.11/site-packages/vorta/store/migrations.py", line 4, in <module>
    from .models import (
  File "/usr/lib/python3.11/site-packages/vorta/store/models.py", line 11, in <module>
    from vorta.utils import slugify
  File "/usr/lib/python3.11/site-packages/vorta/utils.py", line 18, in <module>
    from vorta.borg._compatibility import BorgCompatibility
  File "/usr/lib/python3.11/site-packages/vorta/borg/_compatibility.py", line 1, in <module>
    from pkg_resources import parse_version
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 3327, in <module>
    @_call_aside
     ^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 3302, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 3340, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 633, in _build_master
    return cls._build_from_requirements(__requires__)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 646, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 829, in resolve
    dist = self._resolve_dist(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 870, in _resolve_dist
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'platformdirs<4.0.0,>=2.6.0' distribution was not found and is required by vorta
```

**Comment by CPU-Blanc** [comment]
Exact same issue here on EndeavourOS 6.6.1 Budgie (No point posting logs, it's the exact same stack trace as above)
Confirmed that downgrading `python-platformdirs` to 3.11 fixed the issue

**Comment by m3nu** [comment]
Fixed earlier today in #1843?

**Comment by nedy13** [comment]
I use ArchLinux, too.

Quick and Dirty fix, after I checked the pull request https://github.com/borgbase/vorta/pull/1843.

edit file `/usr/lib/python3.11/site-packages/vorta-0.8.12-py3.11.egg-info/requires.txt`

changed the version number...start vorta again.
`
[:sys_platform != "darwin"]
platformdirs<5.0.0,>=2.6.0
`

**Comment by real-yfprojects** [comment]
Using pipx is also a good solution to dependency conflicts.

**Comment by jonathanjsimon** [comment]
Yes, but the app is supplied as an AUR package, which is certainly easier to manage than pipx, IMHO.

**Comment by m3nu** [comment]
Maybe we remove this version range altogether, if it's not needed any more. Will check why it was added in the first place.

**Comment by Orko79** [comment]
> I use ArchLinux, too.
> 
> Quick and Dirty fix, after I checked the pull request #1843.
> 
> edit file `/usr/lib/python3.11/site-packages/vorta-0.8.12-py3.11.egg-info/requires.txt`
> 
> changed the version number...start vorta again. `[:sys_platform != "darwin"] platformdirs<5.0.0,>=2.6.0`

Works for me too! But I had to change two lines:
```python
[:sys_platform != "darwin"]                                                                                                                             
platformdirs<5.0.0,>=2.6.0  <<<<<<<<<<<<<
secretstorage 
                                                                                                                                                        
[:sys_platform == "darwin"]                                                                                                                             
platformdirs<5.0.0,>=3.0.0  <<<<<<<<<<<<<
pyobjc-core                                                                                                                                             
pyobjc-framework-Cocoa                                                                                                                                  
pyobjc-framework-LaunchServices
```

**Comment by CPU-Blanc** [comment]
This seems to have been fixed as of version `0.8.12-3`

**Comment by tux-wclp** [comment]
hi, I saw that release 0.8.12-3 is now available on AUR. Has this release fixed the python-platformdirs 4.0.0 issue?
For now I still put this library in the ignored packages list and downgraded to version 3.11

(0s = Endeavors)
